### PR TITLE
adding leetcode solution 1008

### PR DESCRIPTION
**Pull request by shivnandk**
## Related Issue

None

## Description

This pull request adds detailed solutions and documentation for LeetCode problems: 1008, 1013, 1026, and 1051.

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

N/A

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [ ] I have followed security best practices in my code changes. (If applicable)

## Additional Context

The solutions provided cover different approaches in Python, Java, and C++ for the specified LeetCode problems.

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [x] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [x] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [x] [Markdown Guide](https://www.markdownguide.org/)
- [x] [MDX Documentation](https://mdxjs.com/docs/)
- [x] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [x] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [x] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [x] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+**Example 1:**
+
+Input: `preorder = [8,5,1,7,10,12]`
+
+Output: `TreeNode{val: 8, left: TreeNode{val: 5, left: TreeNode{val: 1, left: None, right: None}, right: TreeNode{val: 7, left: None, right: None}}, right: TreeNode{val: 10, left: None, right: TreeNode{val: 12, left: None, right: None}}}`
+
+Explanation:
+The constructed binary search tree from the preorder traversal is shown.
+
+![Example 1](images/example1.png)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md:30]: The image is not available. You can remove the image line or upload an image to use.

File: dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+id: construct-binary-search-tree-from-preorder-traversal
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md:1]: before this line add on top ``` --- ```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md (hunk 2 above)
Comment (shivnandk) [reply dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md:1]: added

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**